### PR TITLE
prometheus: 3.0.0 -> 3.0.1

### PR DESCRIPTION
**Pull request by Kranzes**
Diff: https://github.com/prometheus/prometheus/compare/v3.0.0...v3.0.1

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by flokli** [comment]
```
error: hash mismatch in fixed-output derivation '/nix/store/74s6iwll68jnga8gwgp2fpbkm6cd37zy-prometheus-web-ui-3.0.1.tar.gz.drv':
        likely URL: https://github.com/prometheus/prometheus/releases/download/v3.0.1/prometheus-web-ui-3.0.1.tar.gz
         specified: sha256-a3xyStDsutLjYIEm7t3WilmvO36eMHvd4pOtZYYsJCM=
            got:    sha256-MP7B7gVRQnspXLLNmPH8t3DYKbsQkDV9hPJYbM2rB9M=
error: 1 dependencies of derivation '/nix/store/r2iqrznh64fj4wqvjk76bhad8cjy2wiy-prometheus-3.0.1.drv' failed to build
```

File: pkgs/by-name/pr/prometheus/package.nix [hunk 1]
```diff
@@ -32,7 +32,7 @@
 }:
 
 let
-  version = "3.0.0";
+  version = "3.0.1";
   webUiStatic = fetchurl {
     url = "https://github.com/prometheus/prometheus/releases/download/v${version}/prometheus-web-ui-${version}.tar.gz";
     hash = "sha256-a3xyStDsutLjYIEm7t3WilmvO36eMHvd4pOtZYYsJCM=";
```
Comment (flokli) [inline pkgs/by-name/pr/prometheus/package.nix:38]: this needs updating too.

**Comment by flokli** [comment]
another hash mismatch, this time in the go modules hash:

```
hash mismatch in fixed-output derivation '/nix/store/xsk9d6grcwapdc8qi9f1kkljizad3xwn-prometheus-3.0.1-go-modules.drv':
        likely URL: (unknown)
         specified: sha256-MP7B7gVRQnspXLLNmPH8t3DYKbsQkDV9hPJYbM2rB9M=
            got:    sha256-c96YnWPLH/tbGRb2Zlqrl3PXSZvI+NeYTGlef6REAOw=
```

**Comment by Kranzes** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 369347`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>prometheus</li>
    <li>prometheus.cli</li>
    <li>prometheus.doc</li>
  </ul>
</details>

**Review by fpletz — APPROVED** [review]
LGTM and added a fix for the pushgateway test (unrelated to the version bump).

**Pull request merged**